bcda/cclf/parser.go

```diff
@@ -26,6 +26,108 @@ func getCMSID(name string) (string, error) {
 	return parts[1], nil
 }
 
+func CheckIfAttributionCSVFile(filePath string) bool {
+	pattern := `P\.PCPB\.M\d{4}\.D\d{6}\.T\d{7}`
+	filenameRegexp := regexp.MustCompile(pattern)
+	found := filenameRegexp.Match([]byte(filePath))
+	return found
+}
+
+type CSVParser struct {
+	FilePath string
+}
+
+// func getACOConfigs() ([]service.ACOConfig, error) {
+// 	configs, err := service.LoadConfig()
+// 	if err != nil {
+// 		return []service.ACOConfig{}, err
+// 	}
+// 	return configs.ACOConfigs, err
+
+// }
+
+// type AttributionFileRegex struct {
+// 	fileNameReg string
+// 	regMatches  int
+// }
+
+// GetCSVMetadata builds an entry for the cclf_files table, based on the name of the file.
+// The filename regex is part of aco configuration.
+func GetCSVMetadata(path string) (csvFileMetadata, error) {
+	var metadata csvFileMetadata
+	var err error
+
+	// acos, err := getACOConfigs()
+	// if err != nil {
+	// 	return cclfFileMetadata{}, err
+	// }
+
+	// for i, v := range acos {
+	// 	filenameRegexp := regexp.MustCompile(v.AttributionFile.NamePattern)
+	// 	parts := filenameRegexp.FindStringSubmatch(path)
+	// 	if len(parts) == v.AttributionFile.RegParts {
+	// 		metadata, err = validateCSVMetadata(v.AttributionFile, parts)
+	// 		if err != nil {
+	// 			return cclfFileMetadata{}, nil
+	// 		}
+	// 	}
+	// }
+
+	// CSV (not CCLF) file name convention for TCOCMD: P.PCPB.Myymm.Dyymmdd.Thhmmsst
+	tcocmd := `(P|T)\.(PCPB)\.(M)([0-9][0-9])(\d{2})\.(D\d{6}\.T\d{6})\d`
+
+	filenameRegexp := regexp.MustCompile(tcocmd)
+	parts := filenameRegexp.FindStringSubmatch(path)
+
+	metadata, err = validateCSVMetadata(parts)
```

Comment:
I would agree with you there. I kept it in here since it mimics a lot of the logic for the CCLF ingestion (which now that I think about it, could probably be combined) and wanted to keep it consistent. Would it be OK if we keep it in here for the sake of consistency for now?